src/StorageFlattening.cpp

```diff
@@ -202,8 +285,28 @@ class FlattenDimensions : public IRMutator {
         }
         stmt = LetStmt::make(op->name + ".buffer", builder.build(), stmt);
 
-        // Make the allocation node
-        stmt = Allocate::make(op->name, op->types[0], op->memory_type, allocation_extents, condition, stmt);
+        if (hoisted_storages_map.count(op->name) > 0) {
+            HoistedStorageData &hoisted_storage_data = hoisted_storages[hoisted_storages_map[op->name]];
+            vector<Expr> bounded_extents;
+            for (const auto &e : allocation_extents) {
+                Expr expanded_extent = e;
+                // Iterate from innermost outwards
+                for (auto it = hoisted_storages.rbegin(); it != hoisted_storages.rend(); it++) {
```

Comment:
I think this loop should break after index hoisted_storages_map[op->name]

Reply:
Done.